src/Microsoft.Framework.Configuration.Abstractions/IConfigurationRoot.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Configuration
+{
+    public interface IConfigurationRoot : IConfiguration
```

Comment:
Maybe you could just have it be Reload(bool reloadOnlyChildren) or the opposite, just seems a little weird to have 3 interfaces for this